src/storage/src/storage/bidi/redirect.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::google::rpc::Status as RpcStatus;
+use crate::google::storage::v2::{BidiReadObjectRedirectedError, BidiReadObjectSpec};
+use gaxi::grpc::from_status::to_gax_error;
+use prost::Message;
+use std::sync::{Arc, Mutex};
+
+pub fn handle_redirect(
+    spec: Arc<Mutex<BidiReadObjectSpec>>,
```

Comment:
this can be written by multiple threads?

are there N streams per spec? or something else is going on?